wafer/talks/tests/test_views.py

```diff
@@ -427,3 +453,221 @@ def test_author_p_sees_own_talk(self):
         self.assertEqual(response.data['count'], 2)
         self.assertEqual(response.data['results'][0]['title'], "Talk A")
         self.assertEqual(response.data['results'][1]['title'], "Talk P")
+
+
+class TalkViewSetTests(TestCase):
+
+    def setUp(self):
+        create_user('super', True)
+        self.client = SortedResultsClient(sort_key="title")
+        self.client.login(username='super', password='super_password')
+
+    def mk_result(self, talk):
+        def mk_url(talk_url):
+            return {
+                'id': talk_url.id, 'description': talk_url.description,
+                'url': talk_url.url,
+            }
+        return {
+            'talk_id': talk.talk_id, 'talk_type': talk.talk_type,
+            'status': talk.status, 'title': talk.title,
+            'abstract': talk.abstract.raw,
+            'corresponding_author': talk.corresponding_author.id,
+            'authors': [talk.corresponding_author.id],
+            'kv': [],
+            'urls': [mk_url(url) for url in talk.talkurl_set.all()]
+        }
+
+    def test_list_talks(self):
+        talk_a = create_talk("Talk A", ACCEPTED, "author_a")
+        talk_b = create_talk("Talk B", REJECTED, "author_b")
+        response = self.client.get('/talks/api/talks/')
+        result_0, result_1 = response.data['results']
```

Comment:
Why are we splitting this if we're then testing response.data['results'] directly?